smoke-tests/src/test/java/com/splunk/opentelemetry/SmokeTest.java

```diff
@@ -93,9 +95,12 @@ static void setupSpec() {
 
   protected GenericContainer target;
 
-  void startTarget(int jdk) {
+  /** @param targetImageNameSupplier function that returns an image name to start. */
+  void startTarget(Supplier<String> targetImageNameSupplier) {
```

Comment:
Why do you want `Supplier` here? And not just simple `String`?

Reply:
`String` is good. Supplier was a remnant of some intermediary refactoring step. Replaced.